pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -148,66 +152,162 @@ function readProfileFiles(rootPath: string, defaultsPath: string, lockedPath: st
  * @returns null, or the registry data as an object.
  */
-function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY): RecursivePartial<settings.Settings>|null {
+function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY, pathParts: string[]): RecursivePartial<settings.Settings>|null {
   let newObject: RecursivePartial<settings.Settings>|null = null;
 
   const schemaKeys = Object.keys(schemaObj);
+  const schemaKeysLowerCase = schemaKeys.map(k => k.toLowerCase());
+  const commonKeys: string[] = [];
+  const unknownKeys: string[] = [];
+  const specialObjectKeys: string[] = [];
+  let regValue: any;
+
   // ignore case
-  const registryKeys = nativeReg.enumKeyNames(regKey).concat(nativeReg.enumValueNames(regKey)).map(k => k.toLowerCase());
-  const commonKeys = schemaKeys.filter(k => registryKeys.includes(k.toLowerCase()));
+  nativeReg.enumKeyNames(regKey).forEach((k) => {
+    if (k.toLowerCase() === 'integrations' && pathParts.length === 3 && pathParts[2] === 'WSL') {
+      specialObjectKeys.push(k);
+    } else if (k.toLowerCase() === 'extensions' && pathParts.length === 2) {
+      specialObjectKeys.push(k);
+    } else if (k.toLowerCase() === 'mutedchecks' && pathParts.length === 3 && pathParts[2] === 'diagnostics') {
+      specialObjectKeys.push(k);
+    } else if (schemaKeysLowerCase.includes(k.toLowerCase())) {
+      commonKeys.push(k);
+    } else {
+      unknownKeys.push(k);
+    }
+  });
+  if (unknownKeys.length) {
+    console.log(`Unrecognized keys in registry at \\${ pathParts.join('\\') }: [${ unknownKeys.join(', ') }]`);
+  }
 
+  // First process the nested keys, then process any values
   for (const k of commonKeys) {
     const schemaVal = schemaObj[k];
-    let regValue: any = null;
+    const innerKey = nativeReg.openKey(regKey, k, nativeReg.Access.READ);
 
-    if (typeof schemaVal === 'object') {
-      if (!Array.isArray(schemaVal)) {
-        const innerKey = nativeReg.openKey(regKey, k, nativeReg.Access.READ);
+    if (!innerKey) {
+      continue;
+    }
+    try {
+      regValue = readRegistryUsingSchema(schemaVal, innerKey, pathParts.concat([k]));
+    } finally {
+      nativeReg.closeKey(innerKey);
+    }
+    if (regValue) {
+      newObject ??= {};
+      (newObject as Record<string, any>)[k] = regValue;
+    }
+  }
+  for (const k of specialObjectKeys) {
+    const innerKey = nativeReg.openKey(regKey, k, nativeReg.Access.READ);
 
-        if (!innerKey) {
-          continue;
-        }
-        try {
-          regValue = readRegistryUsingSchema(schemaVal, innerKey);
-          if (regValue && (typeof regValue === 'object') && Object.keys(regValue).length === 0) {
-            // Ignore empty objects
-            regValue = null;
-          }
-        } finally {
-          nativeReg.closeKey(innerKey);
-        }
-      } else {
-        const multiSzValue = nativeReg.queryValueRaw(regKey, k);
+    if (innerKey === null) {
+      continue;
```

Comment:
This means, for example, `WSL\intergrations` is a `DWORD` or something like that. Shouldn't that get logged as an error?

Reply:
I agree -- let's make these type mismatches fatal errors.

Reply:
Yes, fixed...